src/esp/gfx/MaterialData.h

```diff
@@ -15,21 +15,42 @@
 namespace esp {
 namespace gfx {
 
-struct MaterialData {};
+struct MaterialData {
+  bool perVertexObjectId = false;
+
+  // construct it using the default constructor. NO initial values, such as
+  // identity matrix
+  Magnum::Matrix3 textureMatrix;
```

Comment:
Right, I see. Identity is a default because it makes more sense than a fully zeroed-out matrix -- if you use a default-constructed matrix in transformations, it's the same as if you didn't use it at all, whereas a zero matrix would wipe them out, which is almost never useful.

Reply:
I will leave it as this.